### ISSUE TITLE
Feature (long term): Using Automotive Templates

### ISSUE THREAD

**Issue by Ixam97** [Ixam97/CarStatsViewer#225]
To enable a public release of the app it es neccessary to implement the Automotive Template Host. 

Polestar won't release the app as a publisher since they no longer plan on releasing any third party apps in the Play Store.

This will require extensive work and testing, which will be done in a seperate branch to not affect the currently established test tracks. Developement will be done in the branch [`playstore_test`](https://github.com/Ixam97/CarStatsViewer/tree/playstore_test)

**Comment by Ixam97** [comment]
The first iteration of the templated version of Car Stats Viewer (for now "Play Edition") has been approved by Google. It is currently available in a closed test. In contrast to internal tests, it has no limit on the number of people participating. It's currently limited to Germany, but will be expanded with the next patch soon (but for now limited to regions with a localization). There will be a link for participation when this patch gets approved.

`playstore_test` has now been merged into `active_development`.

I'll use this issue for further status updates regarding the "Play Edition".

**Comment by Ixam97** [comment]
The patch has been approved. The alpha is now available in the following regions: Australia, Austria, Belgium, Brazil, Canada, Denmark, Finland, France, Germany, Iceland, Ireland, Italy, Liechtenstein, Luxembourg, Netherlands, New Zealand, Norway, Portugal, Spain, Sweden, Switzerland, USA, UK.

For registration, request access to this Google Group: [Car Stats Viewer - Alpha Testing](https://groups.google.com/g/csv-alpha).
Make sure to use the same Google Account as in the car. In the group you will find information on how to enter the closed test. After registration, the App will be available in the Play Store:

![image](https://github.com/Ixam97/CarStatsViewer/assets/12702685/6b85e0b3-751a-499b-ae8c-1290c4ec12fc)

I preserve the right not to accept everybody. Though I plan on moving to an open test sometime soon, as a last step before going into a full release. This open test can then be found directly in the play store without any registration. More on that when the time comes.

Please use this issue for feedback on the Template implementations. All feedback is wellcome as I want to find the best compromise possible.

For bug reports, please open new issues.

**Comment by ivanfmartinez** [comment]
Nice to have this option to access without a many  beta tracks.

Considering this the new main screen I missed the instant information that was available before. As the Volvo driver dashboard does not have numeric scale having an alternate option to see the numbers.


Also to change the trip view now we have to click on "Trip Selection" and choose from a list, on legacy we can just click and "rotate" between the trip types. Maybe the rotation can be made on the top left button when already on the trips "tab"


And now that I have it installed I have completed the pt_BR translation for this version

![20240611_073741](https://github.com/Ixam97/CarStatsViewer/assets/677001/4d924691-6867-494c-bc4c-25a5317820af)

**Comment by Ixam97** [comment]
@ivanfmartinez 

> Considering this the new main screen I missed the instant information that was available before.

In the Settings Tab you can enable real time data. Then you can see power, consumption and altitude in the first Tab.

**Comment by ivanfmartinez** [comment]
By the hint I think that that option was to just increase the frequency of
updates.

Maybe the car_app_show_real_time_data_hint can have a bit more information
about this.

**Comment by DoubleYouEl** [comment]
I’m still running the Dev version next to the Legacy version. When installing the Alpha version, is it best to remove the Dev version?

**Comment by Ixam97** [comment]
@DoubleYouEl You can run them at the same time no problem. They should not interfere eachother. Since you can't transfer data between the versions yet, I would recommend running them at the same time if the data in the Dev-App is relevant for you. If not, you can decide which of them to use/keep.

**Comment by agdejager** [comment]
I like to have the possibilty of making the legacy display as default starting up display. Now you have to search for it and the next time it is gone again. I do not like the new starting up display at all, it is very very dull.

**Comment by jkkataja** [comment]
> 
> And now that I have it installed I have completed the pt_BR translation for this version
> 

Where did you complete it? Previously we've done it in Lokalise but at least I've not received invite to do the translation for this version.

**Comment by Ixam97** [comment]
@agdejager As far as I know this is not possible in templated apps. Other Activities must not be exportet or distraction optimized. I'm not aware of any way to retain the preivous Backstack when opening the App from the Icon. But you can open the App from the notification though and go straight to the legacy dashboard. But keep in mind, it cannot be shown while driving.

**Comment by agdejager** [comment]
OK, if this is the way CSV is going it has died for me. The most important thing for me was to see in real time and graphical how the car was ans is performing. I do not need exporting/importing or all the other things that are in there.

So i will quit the Alpha testing and CSV as a whole.

**Comment by jjtt** [comment]
@agdejager, I suggest you direct your negative feedback to Google and Polestar instead of the developer of this application.

Google Play store sets strict limitations on what apps are allowed to show while driving.

Polestar is able to publish apps that bend the Google rules, but apparently are not interested in publishing 3rd party apps

**Comment by jkkataja** [comment]
Would it be possible to add (Google rule wise and also framerate wise) also the average speed and consumption for last 20/40/100km (or what the legacy selection was again, don't remember) for the new "real time" view?

The bottom red text would be a selector button and the two real time statistics similar to those that are current there.

Or am I alone thinking those would be useful also?

![image](https://github.com/Ixam97/CarStatsViewer/assets/42151530/53ecad8b-4a13-49d4-aa0f-3391f158dfda)

**Comment by Ixam97** [comment]
@jkkataja That is in general a good suggestion. However, the values on the right are already taking the maxium space a custom graphic can take up in that Template. Adding more stuff below that is impossible.

Since I am planning to make the values in the column on the left customizable, it would be possible to add those values for a specific distance to the selectable options. Alternatively, they could be placed in another tab. What tabs are visible could also be made customizable, so you could pick what's important to you and what should only be accessible one layer down in the navigation hierarchy.

**Comment by jkkataja** [comment]
Sounds like a good solution to solve/workaround the limitation.

**Comment by Ixam97** [comment]
There has been an Update to the Automotive Template Host a few days ago. 

The good thing: The API Level was bumped from 6 to 7.

The bad thing: The interval at which I am able to refresh a screens content has increased from 1 to about 7 or 8 seconds. This makes real time data almost unusable. This is nothing I can change by myself and is controlled by Google. As of now, I do not know if this change was inteded or if it is a bug.

**Comment by ivanfmartinez** [comment]
@Ixam97 
I think that this "Error" and "Disabled" are not on Localize

Also will be good to know what was the error , I only found this indication without any detail


![20240626_095539](https://github.com/Ixam97/CarStatsViewer/assets/677001/13e6ddf6-1d4a-4a87-bb2b-2b1aa30d53ab)

**Comment by PolestarEV** [comment]
Just downloaded and installed CSR. Is there a description of the app? Thx

**Comment by PolestarEV** [comment]
> @ivanfmartinez
> 
> > Considering this the new main screen I missed the instant information that was available before.
> 
> In the Settings Tab you can enable real time data. Then you can see power, consumption and altitude in the first Tab.

I see 3 numbers:  power, energy being consumed and the last number, I presume, is altitude…in meters? 

While driving, the numbers seem to updated every ~ 5seconds (so, slight delay). All good. But wondering why they aren’t displayed in graphic format? The legacy format shows various graphs, but that is not available in real time/driving (I’m running the alpha version). I was wonder why that is?

One of the nice things about the Polestar stat viewer (or whatever it’s called) is that it displays a bar graph of the efficiency number, that is updated as you drive (over 20, 40 and 100 miles). I find that type of real time representation of the data to be very useful. 

Any chance that a real time graphical representation of the data will be coming…?

Thx

**Comment by Ixam97** [comment]
@PolestarEV 
> I see 3 numbers: power, energy being consumed and the last number, I presume, is altitude…in meters?
>
> While driving, the numbers seem to updated every ~ 5seconds (so, slight delay). All good. But wondering why they aren’t displayed in graphic format? The legacy format shows various graphs, but that is not available in real time/driving (I’m running the alpha version). I was wonder why that is?

Yes, those are curernt Power, instantaneous consumption and altitude above sea level. There as been an Update to the Automotive App Host. Since then the values cannot be updated in an interval shorter than about 8 seconds. This sadlly is not in my controll and currently the best I can get. 

I'm trying to find other solutions to give more relevant data again, but this kind of stuff is not easy when using the App Host.

The Legacy Dashboard has been the standard while-driving screen in the internal test builds and various tracks created by other users, but cannot be shown while driving in the Play Store Alpha. Thats a hard limit by Google if I want to publish the App publicly via Play Store. Nothing I can do about that I'm afraid.

I have managed to use a Map Template and a custom renderer to display graphs in that, instead of a map. But this also obviously conflicts with Googles Guidelines and will probably not be a viable workaround. But it would look like this: 

<img src="https://github.com/user-attachments/assets/1b64b7f6-5741-42a7-980a-bccf373b1772" width="400"/>

It really is a bummer this kind of stuff is not allowed currently. But on the other hand understandable regarding driver distraction while maintaining a fast turnaround to verify apps.

**Comment by PolestarEV** [comment]
﻿Thank you very much for your detailed response. My comments below:Yes, those are curernt Power, instantaneous consumption and altitude above sea level. There as been an Update to the Automotive App Host. Since then the values cannot be updated in an interval shorter than about 8 seconds. This sadlly is not in my controll and currently the best I can get.
I'm trying to find other solutions to give more relevant data again, but this kind of stuff is not easy when using the App Host.The Legacy Dashboard has been the standard while-driving screen in the internal test builds and various tracks created by other users, but cannot be shown while driving in the Play Store Alpha. Thats a hard limit by Google if I want to publish the App publicly via Play Store. Nothing I can do about that I'm afraid.By AppHost, are you referring to Polestar or Google?Rather than a graph, couldn’t the data be displayed as an update bar graph representation, much like the existing Polestar App, “Range Assistant” does now? That data is displayed in real time (does not blank out when you drive)….so, if there’s no restriction for bar graphs, then maybe this approach might work for you guys? Just a thought? 
I have managed to use a Map Template and a custom renderer to display graphs in that, instead of a map. But this also obviously conflicts with Googles Guidelines and will probably not be a viable workaround. But it would look like this:
image.png (view on web)
Looks good to me….workable approach if allowed? It really is a bummer this kind of stuff is not allowed currently. But on the other hand understandable regarding driver distraction while maintaining a fast turnaround to verify apps.What’s ironic (hypocritical) in this restriction is that things like real time radar weather apps will display while driving….maps constantly update, and drivers are always glancing down to their maps. Why iis that OK, but what you are trying to do is not? I mentioned the Polestar “range assistant” app that displays efficiency data in bar graph format, in real time….that is allowed….so maybe that is a way forward for you guys?Thanks for all your work. George

**Comment by josesampil-cmyk** [comment]
If in the future it is not possible to solve some graphic display like the "Range Assistant" that is already in Polestar 2 while driving, is it possible and of value for users to be able to have a function for graphic visualization, but only when stationary? It may not be feasible, just a thought from me.

**Comment by josesampil-cmyk** [comment]
Is there any possibility to also have access to the previous version with the older dashboard that displays the information graphically and be able to use both versions of the app in parallel without them interfering with each other? I would have appreciated having access to the graphics as I think it is visually very successful, at the same time I would like to be part of the new version and develop it towards a finished product
